packages/core/RNSentryAndroidTester/app/src/test/java/io/sentry/react/RNSentrySDKTest.kt

```diff
@@ -0,0 +1,138 @@
+        mockedRNSentryJsonUtils.close()
+    }
+
+    @Test
+    fun `init with passed configuration callback when no valid json file is provided`() {
+        val mockJsonObject = null
+        mockedRNSentryJsonUtils
+            .`when`<JSONObject> {
+                RNSentryJsonUtils.getOptionsFromConfigurationFile(mockContext, configurationFile, mockLogger)
+            }.thenReturn(mockJsonObject)
+        RNSentrySDK.init(mockContext, mockConfiguration, mockLogger)
+
+        verify(mockLogger).log(
+            eq(SentryLevel.WARNING),
+            contains("Failed to load configuration file(sentry.options.json), starting with configuration callback."),
+        )
+
+        mockedRNSentryStart.verify {
+            RNSentryStart.startWithConfiguration(mockContext, mockConfiguration)
+        }
+    }
+
+    @Test
+    fun `init with passed configuration callback when no valid readable map is created`() {
+        val mockJsonObject = JSONObject()
+        mockedRNSentryJsonUtils
+            .`when`<JSONObject> {
+                RNSentryJsonUtils.getOptionsFromConfigurationFile(mockContext, configurationFile, mockLogger)
+            }.thenReturn(mockJsonObject)
+        val mockReadableMap = null
+        mockedRNSentryJsonUtils
+            .`when`<ReadableMap> {
+                RNSentryJsonUtils.jsonObjectToReadableMap(
+                    mockJsonObject,
+                )
+            }.thenReturn(mockReadableMap)
+
+        RNSentrySDK.init(mockContext, mockConfiguration, mockLogger)
+
+        verify(mockLogger).log(
+            eq(SentryLevel.WARNING),
+            contains("Failed to load configuration file(sentry.options.json), starting with configuration callback."),
+        )
+        mockedRNSentryStart.verify {
+            RNSentryStart.startWithConfiguration(mockContext, mockConfiguration)
+        }
+    }
+
+    @Test
+    fun `init with the json file and the passed configuration when a valid json is provided`() {
+        val mockJsonObject = JSONObject()
+        mockedRNSentryJsonUtils
+            .`when`<JSONObject> {
+                RNSentryJsonUtils.getOptionsFromConfigurationFile(mockContext, configurationFile, mockLogger)
+            }.thenReturn(mockJsonObject)
+        val mockReadableMap =
+            JavaOnlyMap.of(
+                "dsn",
+                "https://abc@def.ingest.sentry.io/1234567",
+            )
+
+        mockedRNSentryJsonUtils
+            .`when`<ReadableMap> {
+                RNSentryJsonUtils.jsonObjectToReadableMap(
+                    mockJsonObject,
+                )
+            }.thenReturn(mockReadableMap)
+
+        RNSentrySDK.init(mockContext, mockConfiguration, mockLogger)
+
+        mockedRNSentryStart.verify {
+            RNSentryStart.startWithOptions(mockContext, mockReadableMap, mockConfiguration, null, mockLogger)
+        }
+    }
+
+    @Test
+    fun `fails with an error when there is an unhandled exception in initialisation`() {
```

Comment:
This test is a bit confusing to me, as the `getOptionsFromConfigurationFile` should never throw.